controllers/workspace/devworkspace_controller.go

```diff
@@ -123,10 +123,13 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return reconcile.Result{}, err
 	}
 
+	configResolveStatus := &workspaceConditions{}
```

Comment:
We could instead init the reconcileStatus struct here:
```go
reconcileStatus := currentStatus{}
// maybe, if warning: 
reconcileStatus.setConditionTrue(conditions.DevWorkspaceWarning, fmt.Sprint("Error applying external DevWorkspace-Operator configuration: ", err.Error()))
```
and potentially set the warning condition in it if necessary. Then later, at lines 214/215
```diff
-	reconcileStatus := currentStatus{phase: dw.DevWorkspaceStatusStarting}
+	reconcileStatus.phase = dw.DevWorkspaceStatusStarting
	reconcileStatus.setConditionTrue(conditions.Started, "DevWorkspace is starting")
```